perl/lib/Gherkin/MarkdownTokenMatcher.pm

```diff
@@ -0,0 +1,345 @@
+package Gherkin::MarkdownTokenMatcher;
+
+use strict;
+use warnings;
+
+my $DEFAULT_DOC_STRING_SEPARATOR = q/^(```[`]*)(.*)/;
+my $KEYWORD_PREFIX_BULLET        = q/^(\\s*[*+-]\\s*)/;
+my $KEYWORD_PREFIX_HEADER        = q/^(#{1,6}\\s)/;
+
+use Class::XSAccessor accessors => [
+    qw/dialect _default_dialect_name _indent_to_remove _active_doc_string_separator _keyword_types
+      _matched_FeatureLine _non_star_step_keywords/,
+];
+
+use Cucumber::Messages;
+use Gherkin::Dialect;
+
+sub new {
+    my ( $class, $options ) = @_;
+    $options->{'dialect'} ||= Gherkin::Dialect->new( { dialect => 'en' } );
+    my $self = bless $options, $class;
+    $self->_default_dialect_name( $self->dialect_name );
+    my @non_star_step_keywords = sort { length($b) <=> length($a) } map {
+        grep { $_ ne '* ' }
+          @{ $self->dialect->$_ }
+    } qw/Given When Then And But/;
+    $self->_non_star_step_keywords( \@non_star_step_keywords );
```

Comment:
The values of the sorted keywords are dependent on the selected language, which means that it is better to move this bit (appending and sorting) to `change_dialect()`. Sorry for not noticing before.

BTW, I don't think that changing the language using the language tag is supported by GherkinInMarkdown, but at least putting it in `change_dialect()` will make the standard and markdown gherkin tokenizers behave the same way. (good for consistency; predictable behavior)

Reply:
I hope you don't mind me addressing my own comment, but I've fixed this comment: I didn't want you to keep having to go back and forth over minor issues.